### PR TITLE
[cmake] Fix search dependency 

### PR DESCRIPTION
**Pull request by greshilov**

### CODE REVIEW

**Comment by greshilov** [comment]
JTALL

**Comment by mpimenov** [comment]
What is the commit that broke it and why did it work before?

**Comment by greshilov** [comment]
It happened after merging `release-83` and `master` (PR)(https://github.com/mapsme/omim/pull/9157).

**Comment by greshilov** [comment]
Looks like smth here: [commit](https://github.com/mapsme/omim/commit/a8d38bb7c90a19585933ae42f9b7c1ee69cec26b)

**Comment by greshilov** [comment]
JTALL

File: track_analyzing/track_analyzer/CMakeLists.txt [hunk 1]
```diff
@@ -21,29 +21,22 @@ omim_link_libraries(
...
-  map
   routing
   traffic
   routing_common
   storage
-  kml
-  editor
   indexer
   platform
   mwm_diff
```
Comment (syershov) [inline track_analyzing/track_analyzer/CMakeLists.txt]: Надо?

Comment (greshilov) [reply track_analyzing/track_analyzer/CMakeLists.txt]: Да, в первом случае `generator` тянет из `storage`:
```
generator::mwm_diff::ApplyDiff
```
А во втором из `libmwm_diff`:
```
divsuf::divsufsort_include_empty
```

File: search/pysearch/bindings.cpp [hunk 2]
```diff
@@ -186,7 +186,7 @@ struct Context
 {
   Context() : m_engine(m_dataSource, CreateCountryInfoGetter(), search::Engine::Params{}) {}
 
-  DataSource m_dataSource;
+  FrozenDataSource m_dataSource;
```
Comment (mpimenov) [inline search/pysearch/bindings.cpp]: add a note or todo here, please

**Comment by mpimenov** [comment]
I'm ok with this as a temporary fix.
It looks like a better one would be to add dead_strip or something similar to the linker flags (see https://github.com/mapsme/omim/blob/05ae6caa244c64c9c0b9dac6fb5ee0ce7394e6b2/common.pri)

**Comment by mpimenov** [comment]
The root cause is fixed in #9177

**Pull request merged**